browser/src/canvas/CanvasSectionContainer.ts

```diff
@@ -594,12 +619,15 @@ class CanvasSectionContainer {
 
 		// Avoid black default background.
 		this.clearCanvas();
+
+		return true;
 	}
 
 	private redrawCallback(timestamp: number) {
 		this.drawRequest = null;
 
-		this.resizeCanvas();
+		if (!this.resizeCanvas())
```

Comment:
So instead of this, we'd have

```
if (this.needsResize && !this.checkCanvasSizeAfterDisplayChanges()) return;

this.resizeCanvas();
```

Reply:
I think current version is safer.

Reply:
In what way? The only other caller of `this.resizeCanvas()` hasn't been changed and I don't think it's clear what a boolean return for `resizeCanvas` actually means or what you should do with it.

On the other hand, the if statement I've suggested (coupled with the suggested rename) would let you know why it's returning here and the other call-site's behaviour then wouldn't be changed (which I believe is unnecessary?)